src/drivers/uavcan/sensors/differential_pressure.cpp

```diff
@@ -0,0 +1,113 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @author Jacob Crabill <jacob@flyvoly.com>
+ */
+
+#include "differential_pressure.hpp"
+
+#include <drivers/drv_airspeed.h>
+#include <drivers/drv_hrt.h>
+#include <lib/ecl/geo/geo.h>
+#include <parameters/param.h>
+#include <systemlib/err.h>
+
+const char *const UavcanDifferentialPressureBridge::NAME = "differential_pressure";
+
+UavcanDifferentialPressureBridge::UavcanDifferentialPressureBridge(uavcan::INode &node) :
+	UavcanCDevSensorBridgeBase("uavcan_differential_pressure", "/dev/uavcan/differential_pressure",
+				   AIRSPEED_BASE_DEVICE_PATH,
+				   ORB_ID(differential_pressure)),
+	_sub_air(node)
+{
+}
+
+int UavcanDifferentialPressureBridge::init()
+{
+	int res = device::CDev::init();
+
+	if (res < 0) {
+		return res;
+	}
+
+	// Initialize the calibration offset
+	param_get(param_find("SENS_DPRES_OFF"), &_diff_pres_offset);
```

Comment:
Reminds me we should expand this to use calibration ids and allow multiple.

Reply:
using a `module.yaml` like battery does? I'm not entirely sure what needs to be added to make that work

Reply:
Ideally we'd have a PX4Airspeed class that would handle all of the get/set/update/etc. calls, then this sensor bridge would operate the same as the baro and mag bridges.  I guess those updates to the sensors pipeline would need to occur at the same time.  I.e. add "SENS_DPRESx_OFF" params or similar.

Reply:
But I believe that's for a later PR.